reserver/reserver_obj.py

```diff
@@ -1,2 +1,100 @@
 # -*- coding: utf-8 -*-
 """Reserver modules."""
+from .reserver_func import does_package_exist, generate_template_setup_py
+from os import environ, path, getcwd, remove
+from shutil import rmtree
+from sys import executable
+from subprocess import check_output, CalledProcessError
+from re import sub
+
+class Uploader:
+    """
+    The Reserver Uploader class reserves a package name by uploading a template repo to pypi account.
+
+    >>> uploader = Uploader(API_TOKEN, is_test_pypi_account = True) # API_TOKEN refers to pypi(or test pypi)'s account api.
+    >>> uploader.upload_to_pypi(PACKAGE_NAME) # uploads package to the given test pypi account.
+    """
+
+    def __init__(self, api_token, test_pypi=False):
+        """
+        Initialize the Reserver Uploader instance.
+
+        :param api_token: pypi account's api token
+        :type api_token: str
+        :param test_pypi: indicates the given api_token is for a test.pypi account or not.
+        :type test_pypi: bool
+        :return: an instance of the Reserver Uploader
+        """
+        self.username = "__token__"
+        self.password = api_token
+        self.test_pypi = test_pypi
+
+        
+    def upload_to_pypi(self, package_name):
+        """
+        Upload a template package to pypi or test_pypi.
+
+        :param package_name: package name
+        :type package_name: str
+        :return: None
+        """
+        if does_package_exist(package_name, self.test_pypi):
+            print("This package already exists in PyPI.")
+            return False 
+        
+        generate_template_setup_py(package_name)
+
+        environ["TWINE_USERNAME"] = self.username
```

Comment:
Please ensure that these two environment variables are deleted or destroyed at the end of their usage.